feat:(client): show-workshop-independent-lower-jaw

ahmaxed opened the pull request:
This PR rolls out the newly AB tested workshop editor layout to all workshops.
Mobile users will see the older layout. It will be replaced with a more mobile friendly version of the new layout in later iterations.

Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

ref #60589


<!-- Feel free to add any additional description of changes below this line -->

ahmaxed committed:
feat/show-workshop-independent-lower-jaw

ahmaxed committed:
fix: tests

ahmaxed committed:
fix: update tests

majestic-owl448 commented:
is the failing test related to this?

ahmaxed committed:
fix: adjust reset-modal and challenge e2e tests

ahmaxed committed:
Merge branch 'main' into feat/show-independent-lower-jaw-workshops

ojeytonwilliams commented:
Sorry for the delay getting to this, @ahmaxed. I'm a little confused, though. Since this is just rolling out the new, already AB tested variant, why is there new code? I would expect to see deletion of the variant that performed poorly, rather than addition of new functionality. I see that there's a lot of test code, and that's fine, but that doesn't account for all of the additions.

ahmaxed commented:
When I rolled out the editor, I came across some glitches and fixed the along the way.

ojeytonwilliams commented:
Okay, could you describe the issues that were fixed or move the fixes into another PR? Otherwise, if there are bugs, it could be tricky to localize them.

ahmaxed commented:
Main fixes in this PR:

1. Fixed lower jaw rendering at the bottom of each editor when multiple editors are present.
2. Corrected tooltip copy.
3. Ensured focus moves to the correct button before showing a hint, after showing a hint, and after tests pass.
4. Added test attributes and classes to help with testing and styling.

ojeytonwilliams commented on e2e/help-button.spec.ts:
```diff
@@ -69,3 +74,19 @@ test.describe('help-button tests for a page with a reset and help button', () =>
 });
+
+test.describe('Desktop help-button tests for a page with a reset and help button', () => {
+  test('should be present after the user checks their code three times', async ({
```
It's always present. Why does mobile need three tries, but desktop not?

ojeytonwilliams commented on e2e/independent-lower-jaw.spec.ts:
```diff
@@ -0,0 +1,188 @@
+    page.getByTestId('independentLowerJaw-submit-button')
+  ).toHaveCount(0);
+});
+
```
```suggestion
```
Let's keep the e2e tests focused on behaviour. Things like "is the button in the page" are better checked incidentally while testing the button's functionality.

ahmaxed committed:
Apply suggestions from code review

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

ojeytonwilliams reviewed (approved):
Just a nit, so I'm approving

ahmaxed committed:
Update help-button.spec.ts

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

ahmaxed committed:
fix:lint
